main.elm

```diff
@@ -30,15 +30,23 @@ type alias Coord =
     ( Int, Int )
 
 
+{-| Dated values
+-}
+type alias Dated a =
```

Comment:
When possible, I avoid records and replace them by data types. Here, it could be
```elm
type Dated a = Dated Time a
```
Dated also serves as a type constructor then:
```elm
line = Dated someDate someString
```